compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -1288,7 +1288,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             )
         });
         let element_tys = tcx.mk_type_list(element_tys_iter);
-        let pat_ty = tcx.mk_ty(ty::Tuple(element_tys));
+        let pat_ty = tcx.intern_tup(element_tys);
```

Comment:
Why is this one named intern_tup instead of mk_tuple? Seems different from the other Ty interners

Reply:
Not sure, it's pre-existing.